d2/src/test/java/com/linkedin/d2/xds/TestXdsClientImpl.java

```diff
@@ -836,46 +867,70 @@ private static final class XdsClientImplFixture
-
+      _nodeWildcardSubscriber = spy(new WildcardResourceSubscriber(NODE));
+      _nodeWildcardSubscriber.setData("fooNode", NODE_UPDATE1);
+      _uriMapWildcardSubscriber = spy(new WildcardResourceSubscriber(D2_URI_MAP));
+      _uriMapWildcardSubscriber.setData(URI1, D2_URI_MAP_UPDATE_WITH_DATA1);
+      _nameWildcardSubscriber = spy(new WildcardResourceSubscriber(D2_CLUSTER_OR_SERVICE_NAME));
+      _nameWildcardSubscriber.setData(CLUSTER_RESOURCE_NAME_IRV, CLUSTER_NAME_DATA_UPDATE);
       doNothing().when(_resourceWatcher).onChanged(any());
       doNothing().when(_wildcardResourceWatcher).onChanged(any(), any());
       doNothing().when(_serverMetricsProvider).trackLatency(anyLong());
 
       for (ResourceSubscriber subscriber : Lists.newArrayList(_nodeSubscriber, _clusterSubscriber, _d2UriSubscriber))
       {
+        ResourceType adjustedType = subscriber.getType();
         _subscribers.get(subscriber.getType()).put(subscriber.getResource(), subscriber);
+        if (adjustedType == D2_URI_MAP && useGlobCollections){
+           adjustedType = D2_URI;
+          _resourceVersions.get(adjustedType).put(
+              GlobCollectionUtils.globCollectionBaseUrlForClusterResource(subscriber.getResource()), "v1");
+        }
+        else {
+          _resourceVersions.get(adjustedType).put(subscriber.getResource(), "v1");
+        }
+
       }
+
       for (WildcardResourceSubscriber subscriber : Lists.newArrayList(_nodeWildcardSubscriber,
           _uriMapWildcardSubscriber, _nameWildcardSubscriber))
       {
-        _wildcardSubscribers.put(subscriber.getType(), subscriber);
+        ResourceType adjustedType = subscriber.getType();
+        _wildcardSubscribers.put(adjustedType, subscriber);
+        if (adjustedType == D2_URI_MAP && useGlobCollections){
+          adjustedType = D2_URI;
+        }
+        ResourceType finalAdjustedType = adjustedType;
+        subscriber.getResourceKeys().forEach(x-> _resourceVersions.get(finalAdjustedType).put(x, "v1"));
       }
 
       _xdsClientImpl = spy(new XdsClientImpl(null, null,
           Executors.newSingleThreadScheduledExecutor(new NamedThreadFactory("test executor")),
-          0, useGlobCollections, _serverMetricsProvider));
+          0, useGlobCollections, _serverMetricsProvider, useIRV));
       _xdsClientImpl._adsStream = _adsStream;
 
       doNothing().when(_xdsClientImpl).startRpcStreamLocal();
       doNothing().when(_xdsClientImpl).sendAckOrNack(any(), any(), any());
-      doNothing().when(_adsStream).sendDiscoveryRequest(_resourceTypesArgumentCaptor.capture(), _resourceNamesArgumentCaptor.capture());
+      doNothing().when(_adsStream).sendDiscoveryRequestWithIRV(_resourceTypesArgumentCaptor.capture(),
+                                                               _resourceNamesArgumentCaptor.capture(),
+                                                               _resourceVersionsArgumentCaptor.capture());
 
```

Comment:
Don't format code like this, it's against the java code guidelines because it doesn't work well when code gets refactored. For example, if the name `sendDiscoveryRequestWithIRV` is ever changed, everything will be misaligned

Reply:
You could format the code with the "LinkedIn Style" saved in the IDE preferences.